Name: Top Bar Ringer Switch
Version: 1.3.5.2-4
Author: frause
Description: Adds icon to top bar when ringer switch is activated. If my Top Bar Icon And Percent Small White patch is installed, the percent value is removed when the mute icon is displayed to reduce witdh of the icon row. (MIT license)


--- /usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js.orig	Fri Jan 15 23:04:41 2010
+++ /usr/lib/luna/system/luna-systemui/app/controllers/bar-assistant.js	Fri Jan 15 23:30:12 2010
@@ -2870,7 +2871,15 @@
 },
 
 audioInit: function() {
-	
+			
+	this.controller.serviceRequest('palm://com.palm.audio/ringtone', {
+		method:'status',
+		parameters: {subscribe:true},
+		onSuccess: function(response) {
+			this.displayMutedIcon(!response['ringer switch']);
+		}.bind(this)
+	});
+
 	//Subscribe to Audio Manager for Phone status Notifications.
 	this.audioPhoneNotificationSession = new Mojo.Service.Request('palm://com.palm.audio/phone', {
 		method: 'status',
@@ -2902,6 +2911,17 @@
 		return;
 			
 	this.handleAudioNotifications(payload,'phone');
+},
+
+displayMutedIcon: function(mute) {
+	if(mute) {
+		$('ringer').className = 'off';
+		$('batpercent').className = 'hide';
+	}
+	else {
+		$('ringer').className = 'on';	
+		$('batpercent').className = 'show';		
+	}
 },
 
 handleMediaAudioNotifications: function(payload) {
--- /usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html.orig	Fri Jan 15 23:17:58 2010
+++ /usr/lib/luna/system/luna-systemui/app/views/bar/bar-scene.html	Fri Jan 15 22:57:11 2010
@@ -19,6 +19,8 @@
 	<div id="devicemenu">
 		<div id="power">
 		</div>
+		<div id="ringer">
+		</div>
 		<div id="rssi">
 		</div>
 		<div id="roaming">
--- /usr/lib/luna/system/luna-systemui/stylesheets/systemui.css.orig	Fri Jan 15 23:20:50 2010
+++ /usr/lib/luna/system/luna-systemui/stylesheets/systemui.css	Fri Jan 15 23:31:21 2010
@@ -122,6 +122,7 @@
 #tty,
 #hac,
 #callforward,
+#ringer,
 #gps {
 		float: right;
 		height: 20px;
@@ -217,6 +218,21 @@
 	background: url(../images/battery-charged.png) center center no-repeat;
 }
 
+#ringer {
+	width: 20px;
+	margin-left: 2px;
+}
+#ringer.off {
+	background: url(../../../../../palm/applications/com.palm.app.clock/images/prefs-mute.png) center left no-repeat;
+}
+#ringer.on {
+	display: none;
+}
+#batpercent.hide {
+	display: none;
+}
+#batpercent.show {
+}
 
 #rssi {
 	width: 19px;
